ocs_ci/ocs/ui/base_ui.py

```diff
@@ -0,0 +1,178 @@
+
+    def is_expanded(self, by_locator, type=By.XPATH, timeout=30):
+        """
+        Check whether an element is in an expanded or collapsed state
+
+        Args:
+            by_locator (str): GUI element needs to operate on.
+            type (By): Set of supported locator strategies
+            timeout (int): Looks for a web element repeatedly until timeout (sec) happens.
+
+        return:
+            bool: True if element expended, False otherwise
+
+        """
+        wait = WebDriverWait(self.driver, timeout)
+        element = wait.until(ec.element_to_be_clickable((type, by_locator)))
+        return True if element.get_attribute("aria-expanded") == "true" else False
+
+    def choose_expanded_mode(self, mode, by_locator, type):
+        """
+        Select the element mode (expanded or collapsed)
+
+        by_locator (str): GUI element needs to operate on.
+        type (By): Set of supported locator strategies
+        mode (bool): True if element expended, False otherwise
+
+        """
+        current_mode = self.is_expanded(by_locator=by_locator, type=type)
+        if mode != current_mode:
+            self.do_click(by_locator=by_locator, type=type)
+
+    def take_screenshot(self):
+        """
+        Take screenshot using python code
+
+        """
+        time.sleep(2)
+        self.cnt_screenshot += 1
+        filename = os.path.join(
+            self.screenshots_folder, str(self.cnt_screenshot).zfill(3)
+        )
+        self.driver.save_screenshot(filename)
+
+
+def login_ui():
+    """
+    Login to OpenShift Console
+
+    return:
+        driver (Selenium WebDriver)
+
+    """
+    logger.info("Get URL of OCP console")
+    console_url = run_cmd(
+        "oc get consoles.config.openshift.io cluster -o"
+        "jsonpath='{.status.consoleURL}'"
+    )
+    logger.info("Get password of OCP console")
+    password = get_kubeadmin_password()
+    password = password.rstrip()
+    browser = ocsci_config.UI_SELENIUM.get("browser_type")
+    if browser == "chrome":
+        logger.info("chrome browser")
+        chrome_options = Options()
+
+        ignore_ssl = ocsci_config.UI_SELENIUM.get("ignore_ssl")
+        if ignore_ssl:
+            chrome_options.add_argument("--ignore-ssl-errors=yes")
+            chrome_options.add_argument("--ignore-certificate-errors")
+
+        # headless browsers are web browsers without a GUI
+        headless = ocsci_config.UI_SELENIUM.get("headless")
+        if headless:
+            chrome_options.add_argument("--headless")
+
+        chrome_browser_type = ocsci_config.UI_SELENIUM.get("chrome_type")
+        if chrome_browser_type == "chromium":
+            chrome_browser = ChromeType.CHROMIUM
+
+        driver = webdriver.Chrome(
+            ChromeDriverManager(chrome_type=chrome_browser).install(),
+            chrome_options=chrome_options,
+        )
+    if browser == "firefox":
+        logger.info("firefox browser")
+        driver = webdriver.Firefox()
```

Comment:
Is Firefox actually fully supported? If not, it might be better to raise some exception with description, that execution with Firefox is not yet fully implemented.